lib/task_bunny/worker.ex

```diff
@@ -50,14 +52,18 @@ defmodule TaskBunny.Worker do
     start_link(%Worker{host: host, job: job, concurrency: concurrency})
   end
 
-  @doc false
+  @doc """
+  Starts a worker given a worker's state
+  """
+  @spec start_link(%Worker{}) :: GenServer.on_start
```

Comment:
can you use `t` or `Worker.t` instead of `%Worker{}`? Same can be applied to other specs below.